sdk/metrics/src/jmh/java/io/opentelemetry/sdk/metrics/aggregator/BucketSearch.java

```diff
@@ -0,0 +1,244 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.metrics.aggregator;
+
+import java.util.Arrays;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.concurrent.TimeUnit;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.Measurement;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.Threads;
+import org.openjdk.jmh.annotations.Warmup;
+
+@State(Scope.Benchmark)
+public class BucketSearch {
```

Comment:
what is the purpose of this benchmark? It doesn't seem to be benchmarking anything but how fast we can loop through an array. And, it does it with randomly selected values, which I'm not sure will end up providing a useful benchmark for anything.

Reply:
The purpose is to justify the linear search of `DoubleHistogramAggregator.Handle.State.findBucketIndex`.

Here's the basic setup:
- the bucket boundaries are grabbed from some real cases.
- the values are selected in a uniform-distribution with proper upper bound related to the boundaries.
    - in real world, most value distribution would bias toward to the left end. so the results would serve as an estimated upper bound of the performance.

but I'm not sure if I should put the benchmark in the PR actually since it's not too general. Do u suggest removing this file?

Reply:
Yes, I don't think that benchmarking linear searching, out of context of the actual usage of that search, has much value. I would definitely recommend removing it from this PR.